### PR TITLE
To Do list: list structure

### PR DESCRIPTION
**Pull request by SlowKingV**
## Description
In this project, you will build a simple HTML list of *To Do* tasks. The list will be styled according to the specifications listed later in this lesson. This simple web page will be built using webpack and served by a webpack dev server.

## Project requirements
- Set up a new project with webpack that is based on the webpack exercise you have already completed.
- Create an **index.html** file and write your HTML markup here. Create an empty **To Do List** placeholder (`<div>` or `<ul>` element). The **index.html** file must be set as a template using the *HTML Webpack Plugin*.
- Create an **index.js** file and set an array of some simple *to do tasks* (array of objects). Each task object should contain three keys: 
  1. description [`string`].
  2. completed [`bool`].
  3. index: [`number`].
- Write a function to iterate over the tasks array and populate an HTML list item element for each task.
- On page load render the dynamically created list of tasks in the dedicated placeholder. The list should appear in order of the `index` values for each task.
- Create a **style.css** and set rules for the **To Do List**. CSS must be loaded by *Webpack Style/CSS Loader*. Your list should be a clone of the part of the minimalist project captured in the video below.
- All your source files (**index.html**, **index.js** and **style.css**) must be located in */src* directory and your distribution files will be generated by webpack and served by *webpack dev server* from */dist* folder.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1 +1,57 @@
...
+## Built With
+
+- HTML, CSS, JavaScript
+- Used [Webpack](https://webpack.js.org/) as bundler
+- Icons from [FontAwesome](https://fontawesome.com/)
+- (Used Lighthouse, Webhint, Stylelint, and ESLint as linters)
+
+## Live Demo
+
+You can check a live working version of this project here: [live-demo](https://slowkingv.github.io/todo-list/)
```
Comment (codecaiine) [inline README.md:16]: - [x] Good job, your readme looks descriptive and professional  :+1: unfortunately your live demo is not working :cry: . Please fix it in order to help people see your project live and keep your readme more professional :smile:

File: src/index.js [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+// On page load populate task list
+window.onload = () => {
+  const listContainer = document.getElementById('list');// <ul> element to put the <li>s
+  const tasklist = new TaskList();
+  tasklist.add('Task 0');
+  tasklist.add('Task 1');
+  tasklist.add('Task 2');
+  tasklist.add('Task 3');
+  tasklist.insertItems(listContainer);
+};
```
Comment (codecaiine) [inline src/index.js]: - [x] Good job so far :smile:  . But Please check from the project requirements, on page load, the task array is rendered dynamically to the DOM. The tasks should appear in order of the index values for each task. for example, if I change the order of the index, it should update in the DOM.  ️[Check the project requirements](https://github.com/microverseinc/curriculum-javascript/blob/main/todo-list/m1_list_structure.md)️. Please you can notice below that, this is one way you can implement this :point_down: 

#### Example  *to do tasks* (array of objects)

You array of some simple to-do tasks (array of objects) with each object containing three keys - **description [string], completed [bool], index: [number].**  ....
```
const todoList = [{
  description: 'Go to the market',
  completed: false,
  index: 4,
},
....
]

```

#### example of a function that iterates over the tasks array and populates the HTML list item element for each task.

```
const showTasks = () => {
  for (let i = 0; i < todoList.length; i += 1) {
    const task = todoList[i];
    const list = ` <li class="todo" id="${task.index}">
      <input type="checkbox" class="check" id="list-checkbox" name="list-checkbox">
      ${task.description}<span class="icon"><i class="fa fa-ellipsis-v"></i></span> <span class="btn-del"><i class="fa fa-trash"></i></span>
  </li>`;
  }
};

```

**Review by codecaiine — CHANGES_REQUESTED** [review]
Hi @SlowKingV , 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights
- Descriptive Pull request :heavy_check_mark: 
- Good use of Gitflow :heavy_check_mark: 
- Commits are meaningful :heavy_check_mark: 

## Required Changes ♻️

_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me (@codecaiine) in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: README.md (hunk 1 above)
Comment (SlowKingV) [reply README.md:16]: I've renamed the build output folder to 'docs/' so now the live demo should be working!

File: src/index.js (hunk 2 above)
Comment (SlowKingV) [reply src/index.js]: I've created two new methods:

1. The first one is `moveToPosition(taskIndex, newIndex)` (inside TaskList) and it allows you to move a task to a different index in the array.
https://github.com/SlowKingV/todo-list/blob/71b78a410327f60c5481ec9b98b6568b0587d1ab/src/modules/taskList.js#L18-L22
2. The second one is `updateIndex()` (inside Task) which reflects index changes into the HTML `<li>` element, as a custom attribute called `data-index`.
https://github.com/SlowKingV/todo-list/blob/71b78a410327f60c5481ec9b98b6568b0587d1ab/src/modules/task.js#L9-L11

**Comment by SlowKingV** [comment]
All requested changes are finished and everything should be working as expected!
I also added some example tasks and operations here:
https://github.com/SlowKingV/todo-list/blob/71b78a410327f60c5481ec9b98b6568b0587d1ab/src/index.js#L15-L25

**Review by Reem-lab — APPROVED** [review]
# Hi @SlowKingV, 👋🏻👋🏻 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 🔥🔥🔥💯💯


## Highlights
 - Good Commits messages ✅✅
 - Descriptive pull request and branch name ✅✅
 - No Linters error✅✅

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

**Pull request merged**